creator-node/test/CNodeHealthManager.test.js

```diff
@@ -0,0 +1,478 @@
+    // Stub functions that isNodeHealthy() will call
+    const node = 'http://some_content_node.co'
+    const verboseHealthCheckResp = { healthy: true }
+    const queryVerboseHealthCheckStub = sandbox
+      .stub(CNodeHealthManager, 'queryVerboseHealthCheck')
+      .resolves(verboseHealthCheckResp)
+    const determinePeerHealthStub = sandbox.stub(
+      CNodeHealthManager,
+      'determinePeerHealth'
+    )
+    const logErrorStub = sandbox.stub(CNodeHealthManager, 'logError')
+
+    // Verify that only the simple check was performed and returned healthy
+    const isHealthy = await CNodeHealthManager.isNodeHealthy(node)
+    expect(isHealthy).to.be.true
+    expect(queryVerboseHealthCheckStub).to.have.been.calledOnceWithExactly(node)
+    expect(determinePeerHealthStub).to.have.been.calledOnceWithExactly(
+      verboseHealthCheckResp
+    )
+    expect(logErrorStub).to.not.have.been.called.called
+  })
+})
+
+describe('test CNodeHealthManager -- queryVerboseHealthCheck()', function () {
+  let sandbox
+  beforeEach(function () {
+    sandbox = sinon.createSandbox()
+    nock.disableNetConnect()
+  })
+
+  afterEach(function () {
+    sandbox.restore()
+    nock.cleanAll()
+    nock.enableNetConnect()
+  })
+
+  it('returns successful response', async function () {
+    const endpoint = 'http://healthy_cn.co'
+    const verboseHealthResp = {
+      healthy: 'true',
+      verboseData: 'data'
+    }
+    nock(endpoint)
+      .get('/health_check/verbose')
+      .reply(200, { data: verboseHealthResp })
+
+    await CNodeHealthManager.queryVerboseHealthCheck(endpoint)
+    expect(nock.isDone()).to.be.true
+  })
+})
+
+describe('test CNodeHealthManager -- determinePeerHealth()', function () {
+  // Set config vars for health thresholds
+  const minimumStoragePathSize = 100
+  const minimumMemoryAvailable = 100
+  const maxFileDescriptorsAllocatedPercentage = 50
+  const minimumDailySyncCount = 3
+  const minimumRollingSyncCount = 5
+  const minimumSuccessfulSyncCountPercentage = 50
+  config.set('minimumStoragePathSize', minimumStoragePathSize)
+  config.set('minimumMemoryAvailable', minimumMemoryAvailable)
+  config.set(
+    'maxFileDescriptorsAllocatedPercentage',
+    maxFileDescriptorsAllocatedPercentage
+  )
+  config.set('minimumDailySyncCount', minimumDailySyncCount)
+  config.set('minimumRollingSyncCount', minimumRollingSyncCount)
+  config.set(
+    'minimumSuccessfulSyncCountPercentage',
+    minimumSuccessfulSyncCountPercentage
+  )
+
+  function determinePeerHealth(verboseHealthCheckResp) {
+    const CNodeHealthManagerMock = proxyquire(
+      '../src/services/stateMachineManager/CNodeHealthManager.js',
+      {
+        './../../config': config
+      }
+    )
+    CNodeHealthManagerMock.determinePeerHealth(verboseHealthCheckResp)
+  }
+
+  it("doesn't throw if all data is healthy (empty data counts as healthy)", function () {
+    expect(() => determinePeerHealth({})).to.not.throw()
```

Comment:
I actually don't think we'll encounter this scenario, cus either the verbose health check will return `{..lots of stuff}` or an error will be thrown

Reply:
this is more to check the default happy path where nothing causes it to throw. it has everything as empty because `determinePeerHealth` checks each property's existence first -- for example:
```js
storagePathSize &&
      storagePathUsed &&
      storagePathSize - storagePathUsed <= MINIMUM_STORAGE_PATH_SIZE
```
I can update the test name to say "happy path" or something